docs/adrs/waitlists_relationships.md

```diff
@@ -0,0 +1,344 @@
+# Waitlists as Relationships
```

Comment:
Do you want to add the other ADR made in the other PR; even if it was maybe not used?
https://github.com/mozilla-it/ctms-api/blob/e9e216325b15cabbf7f0c0e9e2417925f61df47f/docs/adrs/basket-inbox.md

Reply:
Do you think it adds value? We would its status as rejected?

Reply:
In the project with `rejected` might make sense; I think there's some value in the adr that hasn't been extracted perhaps as well, since we're still in a blurry boundary with mapping crossover in the basket code:
https://github.com/mozmeao/basket/blob/965c4bd745fd6fe3ccda7495d13a9c2036f627ab/basket/news/backends/ctms.py#L31-L87

I realize maybe now that this conversation is adjacent to this PR...